tests/test_routes.py

```diff
@@ -111,6 +120,49 @@ def test_read_shopcarts(self):
         self.assertEqual(resp.status_code, status.HTTP_200_OK)
         data = resp.get_json()
         self.assertEqual(data["user_id"], shopcart.user_id)
+    
+    def test_update_shopcart(self):
```

Comment:
Same, the old products should be deleted. Your method now is more like a bulk-add instead of bulk-update